### DISCUSSION TITLE
Nested array elements?

### DISCUSSION THREAD

**Discussion by robomac** [halaxa/json-machine#118] [Q&A]
The JSON is of the general form of:
`{
     "foos":[
          { "fooName":"name", "bars" : [
               {"barName":"some other name", "barValue": "value of bar"}, 
               {"barName":"second other name", "barValue": "second value of bar"}
          ]}
   ]
} `

I can iterate through foo with fooName via:
`$items = Items::fromFile($jsonfname, ['pointer' => '/foos']);`
But while bars is listed, it is an empty stdClass.

I tried, per the docs, 
`$items = Items::fromFile($jsonfname, ['pointer' => ['/foos','/foos/bars']]);`
which gave the error:
```
JsonMachine Exception InvalidArgumentException 10.48.25
JSON Pointers must not intersect. At least these two do: '/foos/bars', '/foos'

```
How can I get /foos/bars fully resolved/filled-in?

Thanks

(Note: Seems related to https://github.com/halaxa/json-machine/pull/36 but I don't see what that resolution was.)

**Comment by robomac** [comment]
With a few more hours of experiementation, the answer appears to be... another foreach loop...
`
   $items = Items::fromFile($jsonfname, ['pointer' => '/foos']);
   foreach ($items as $key => $data) {
         foreach ($data->bars as $bar) {
              ... use bar here....
`

**Reply by halaxa** [reply]
If a nested `foreach` works for you memory-wise, it's the best solution.

**Reply by halaxa** [reply]
#36 was merged and released. See Recursive iteration in readme.
